src/components/badges/types.ts

```diff
@@ -3,6 +3,8 @@ import { ReactNode } from "react"
 import { Language } from "../../types/language"
 
 export type BadgeSize = "small" | "large"
+export type BadgeHeight = "small" | "large"
+export type BadgeBackground = "bg-white" | "bg-whatsapp"
```

Comment:
How about

```suggestion
export type BadgeBackground = "white" | "green"
```
you can map the property to the concrete colour in the component, no need to do that in the property value. What do you think?

Reply:
I would push it even further here. The style should be contained with the badge and not passed in from the outside. Is there a way to add the bg color on a wrapper instead of the badge itself?

Reply:
I'd have to add other styles to the wrapper as well, such as rounded corners, borders, set the width of the wrapper... then we'd have to change change the wrapper each time we want something different, instead of sending a prop, which feels weird to me. I feel it's easier to understand and do with it being in one div

also, can you elaborate on the downside of passing it from the outside? This is internal to the components-pkg, and can't be passed from listings and DH

Reply:
> Generally, we aim to be quite on point with the interfaces (ie. props) of shared components in components-pkg.

Agreed! It's worth spending extra time on time, as fixing is later is more time-consuming than getting it right the first time.

> We aim for clear and straight forward interfaces that don't allow any weird combinations that should never happen according to the styleguide.

I took extra care that externally weird combinations cannot be passed as props. So, we cannot have a green money back guarantee badge if it's design is intended to be white. Internally the combinations looked weird because of the naming. I tried to rename it in a fix. Can you take a second look?

> BadgeBackground is currently a css class, which I believe is a violation of proper encapsulation.

I didn't know this! Thanks for the detailed explanation. I was really going off of how somebody corrected me in a review a while back. 

> The badgesize/badgeheight are confusing. I wouldn't really know what effect they have just by looking at the names. I'm also not sure if all combinations would be allowed (small height, large size for example). 

This has been addressed, but the internal mapping to me now looks weird, which is why I suggested we fix it in a follow-up tech ticket anyway. Now externally we pass `size="large"`, but internally `size` maps to `withText`, and to make things more confusing, `size` is also used internally, but in this case it is actually the size (as in width and height).

Reply:
Ok, if you're all comfy, let's go with this. A lot of this might be preference / style and I don't want to impose too much. Especially if this is an internal component